app/controllers/PassMarkSettingsController.scala

```diff
@@ -18,36 +18,61 @@ package controllers
 
 import factories.UUIDFactory
 import model.Commands.Implicits._
-import model.exchange.passmarksettings.Phase1PassMarkSettings
+import model.exchange.passmarksettings.{ PassMarkSettings, Phase1PassMarkSettings, Phase2PassMarkSettings, Phase3PassMarkSettings }
 import org.joda.time.DateTime
-import play.api.libs.json.Json
+import play.api.libs.json.{ Format, Json }
 import play.api.mvc.Action
 import services.AuditService
-import services.passmarksettings.{ PassMarkSettingsService, Phase1PassMarkSettingsService }
+import services.passmarksettings._
 import uk.gov.hmrc.play.microservice.controller.BaseController
 
 import scala.concurrent.ExecutionContext.Implicits.global
 
-object Phase1PassMarkSettingsController extends Phase1PassMarkSettingsController {
+object Phase1PassMarkSettingsController extends PassMarkSettingsController[Phase1PassMarkSettings] {
   val passMarkService = Phase1PassMarkSettingsService
   val auditService = AuditService
   val uuidFactory = UUIDFactory
+  val passMarksCreatedEvent = "Phase1PassMarkSettingsCreated"
+  def getCopy(passMarkSettings:Phase1PassMarkSettings, newVersionUUID: String) =
+    passMarkSettings.copy(version = newVersionUUID, createDate = DateTime.now())
 }
 
-trait Phase1PassMarkSettingsController extends BaseController {
+object Phase2PassMarkSettingsController extends PassMarkSettingsController[Phase2PassMarkSettings] {
+  val passMarkService = Phase2PassMarkSettingsService
+  val auditService = AuditService
+  val uuidFactory = UUIDFactory
+  val passMarksCreatedEvent = "Phase2PassMarkSettingsCreated"
+  def getCopy(passMarkSettings:Phase2PassMarkSettings, newVersionUUID: String) =
```

Comment:
Some comment
[I guess it is not a copy if you change the version, does it sound better to say "upgradeVersion" (if that is the intention)?]